src/test/compile-fail-fulldeps/proc-macro/span-api-tests.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// aux-build:span-api-tests.rs
+
+#![feature(proc_macro)]
+
+extern crate span_api_tests;
+
+use span_api_tests::{reemit, assert_no_source_file};
+
+reemit! {
+    assert_no_source_file! { "Hello, world!" }
```

Comment:
Re-parsing from a string creates tokens with the macro invocation span and then applies a mark to indicate that they are macro-expanded. It doesn't create a "fake" file, so that shouldn't be used to determine whether something came from a macro or not.